### ISSUE TITLE
Iphone x grey screen tone

### ISSUE THREAD

**Issue by italidisco** [checkra1n/BugTracker#1130]
Tell us about your setup:

What iDevice are you using? iPhone x
On what version of iOS is it? 13.4
What version of checkra1n are you using, and GUI or CLI? 0.10.0 beta
What is your host system (OS version? Hackintosh? VM? etc.)?
How are you connecting to the device (USB-A? USB-C? Apple/3rd party cable? Through a USB hub?)?
What are the steps to reproduce the issue?

unlock screen
...

What do you expect, and what is happening instead?

When i unlock screen, many times the screen take a grey tone and after 2/3 seconds the screen turns ok.
Not visible when taking a screenshot.

Does the issue also occur if you enable Safe Mode?

Any other info, error logs, screenshots, ...?

Bug not visible in screenshot

**Comment by Siguza** [comment]
And that's only happening while jailbroken? Because it sounds like a hardware issue, the way you describe it...

**Comment by italidisco** [comment]
No. Is not a hardware problem.
Is a problem related with jb ( checkra1n or some tweak/s).
I restarted phone (loosing jb) and the problem is solved.

I am surprised not to have seen anyone else with this problem and so I ask, in case there is evidence of this failure or otherwise investigate the tweaks

**Comment by yur1xpp** [comment]
@Siguza I'm having the **exact same issue**. Screenshot won't capture the color. It looks like a layer of UIView on top of some UI elements, and after few seconds, it disappears. My most reproducible way of this issue is put the phone on a wireless charging for 10-15 minutes, and after that take the phone off the wireless charging and immediately open app (that was killed priory before put in wireless charging) that has dark mode ability, like Preferences. From there, you can see the rows are greenish/grey in colour for few seconds before return to the original color. This might be a daemon issue? This issue never occurred to me on iOS 13.3.1 using checkra1n version 0.9.8.2, with the exact same tweaks configuration.

Other users that having this issue: https://www.reddit.com/r/jailbreak/comments/fzonnn/discussion_some_ui_elements_becomes_greenish_dark/

The guy even did a fresh reinstall of iOS on iOS 13.3 and the issue still persist, and my best bet is he's using the latest checkra1n too, awaiting for his response on reddit.

iPhone X, iOS 13.4.1, checkra1n 0.10.1.

**EDIT**: It doesn't necessary needs to be on wireless charging, just tested that as long as the device is in deep sleep for 10-15 minutes, this issue pretty much will occur.

**Comment by yur1xpp** [comment]
**UPDATE**: It occurred in Safe Mode entered via checkra1n itself. This definitely means this is a checkra1n issue on version 0.10.1. @Siguza

**Comment by yur1xpp** [comment]
I captured the video here: https://imgur.com/a/3nKHxiu
Notice the color of the row in Preferences changes near the end of the video. I chose the Preferences app because it's easier to capture, it happens everywhere for the record.  Excuse the masking.

**Comment by italidisco** [comment]
Yes, the same.
Is an issue with last checkra1n version.
Im thinking unjailbreak until this issue be solved (if is solved)

**Comment by yur1xpp** [comment]
Now it's becoming worst, it stays there for very long. Even ldrestart didn't make it go away, only reboot will make it go away.

**Comment by p2kdev** [comment]
I am seeing the same behavior. iPX on 13.4

**Comment by italidisco** [comment]
Apparently, with the last version of checkra1n, i not have the problem (for now...)

**Comment by yur1xpp** [comment]
@italidisco what do you mean?

**Comment by italidisco** [comment]
I thought it was already fixed but it continues to do so ...

**Comment by italidisco** [comment]
Im now on 13.5 and have the same problem, but less frequently.

I will try the new unc0ver jailbreak today...

**Comment by yur1xpp** [comment]
Yep, can confirm this issue persist in 13.5 and checkra1n 0.10.2.

@italidisco did you try unc0ver? I wonder if this issue also happens in unc0ver.

**Comment by italidisco** [comment]
And with unc0ver the problem exists too..  but less frequently.
Is not a problem with checkra1n in particular. Is a problem with some jailbreak file..

**Comment by yur1xpp** [comment]
Could you file a bug report to pwn too? Maybe he'll have a look and we'll solve this eventually.

**Comment by italidisco** [comment]
I do not know the truth. English is not my first language and I cannot express myself as I would like ..
 what is strange to me is that nobody complains about the same problem neither on Twitter nor anywhere.

**Comment by yur1xpp** [comment]
I think it’s because it’s a very subtle change of the color, they just didn’t notice about it, since it disappear a few seconds later. Since this issue occur in Safemode in checkra1n (not `killall -SEGV SpringBoard`), it definitely rules out tweak as culprit. @Siguza i was wondering if any member of the checkra1n team having this issue too?

**Comment by Siguza** [comment]
AFAIK very few people on the checkra1n team has an iPhone X.

Also, right now I'm considering this as possibly a valid bug, but if we wanna get anywhere, we need steps to reproduce.

**Comment by yur1xpp** [comment]
The 100% reproduce-able steps of this issue for me is to lock device and wait for 15 minutes then quickly unlock. If you mean more technical way like log or something, so far I can't figure out anything yet, since it occur for only brief moment after unlocking, it's quite hard to trace it.

**Comment by italidisco** [comment]
You can record a video if you have another device with cam.
I would but I have removed the jailbreak.

**Comment by yur1xpp** [comment]
I did in my previous comment: https://github.com/checkra1n/BugTracker/issues/1130#issuecomment-613939409

**Comment by Siguza** [comment]
You can see absolutely nothing there though. Auto brightness makes it look completely normal.

**Comment by yur1xpp** [comment]
Yeah it's a little hard to capture since camera sensor will adjust the exposure, but auto brightness won't have such a sudden change, it's usually gradual. The rows color changing color is actually what the naked eyes see. I've tried screen recording, interesting part is that it won't capture the "grey/greenish tone", the screen brightness stays the same even though the tone is disappearing.

**Comment by italidisco** [comment]
honestly if someone from the staff had iPhone x ... they had already noticed it because it shows and a lot ...

**Comment by yur1xpp** [comment]
This is another user having the issue few days ago: https://www.reddit.com/r/jailbreak/comments/gjmr5z/discussion_appswitcher_lags_weird_screen_colour/
The steps to reproduce this is the same as how I described it.

All of this issue occurred on iPhone X model. There must be something causing this, perhaps the OLED display? Some kind of daemon that late in waking up for the OLED display?

@italidisco I think if we want to solve this issue, we needs to work with the checkra1n team

**Comment by yur1xpp** [comment]
Update: Just did a fresh restore, and this issue persist.

**Comment by yur1xpp** [comment]
@Siguza @italidisco 
https://9to5mac.com/2020/06/05/iphone-11-customers-noticing-random-green-display-bug-unclear-how-to-fix/
I'm so pissed right now, all this time I thought this is a jailbreak issue and try to diagnose it however I could. Any idea on the technicality of this issue, maybe a tweak could fix it?
I must also mention that while in non-jailbroken mode, I personally didn't notice the tint issue, but extremely frequent in jailbroken mode, maybe because I didn't stay long in non-jailbroken mode. Perhaps jailbreak catalyzed the issue? The article suggest it's an iOS 13.5 bug, but I've experienced this since iOS 13.4, in jailbroken mode.
